File: WebSite/EulerAList/pset16.cpp
#include <iostream>
#include <string>

using namespace std;

int main()
{
    string value = "1";
    for (int i = 0; i < 1000; i ++)
    {
        int adder = 0;
        for (long j = value.size() - 1; j >= 0; j --)
        {
            int x = value[j] - '0';
            x = x * 2 + adder;
            value[j] = (char)(x % 10 + '0');
            adder = x / 10;
        }
        value = to_string(adder) + value;
    }
    int res = 0;
    for (char i : value)
    {
        res += i - '0';
    }
    cout << res << endl;
}

File: WebSite/EulerAList/pset16.py
value = "1"
for i in range(1000):
    adder = 0
    tmp = ""
    for j in range(len(value) - 1, -1, -1):
        x = int(value[j])
        x = x * 2 + adder
        tmp = str(x % 10) + tmp
        adder = x // 10
    value = tmp
    value = str(adder) + value
res = 0
for i in value:
    res += int(i)
print(res)